src/other/Table.js

```diff
@@ -307,27 +325,40 @@
             .append("td")
         ;
         cells[this.renderHtmlDataCells() ? "html" : "text"](function (d, idx) { 
-            var retVal = context.columnPatterns()[idx] ? context.getColumnFormatting(d, idx) : d;
+            var retVal;
+            if (!(d instanceof Widget)) {
+                retVal = context.columnPatterns()[idx] ? context.getColumnFormatting(d, idx) : d;
+            }
             return retVal; 
         });
         cells.exit()
             .remove()
-        ;      
+        ;
+
         rows.each(function(tr,trIdx){
-            d3.select(this).selectAll("td").each(function(tdContents,tdIdx){
+            d3.select(this).selectAll("td").each(function(tdContents,tdIdx){    
                 var alignment = context.getColumnAlignment(tdContents);
                 d3.select(this)
-                    .classed("tr-"+trIdx+"-td-"+tdIdx,true)
                     .style("text-align", alignment)
+                    .classed("tr-"+trIdx+"-td-"+tdIdx,true)
                 ;
+                if (tdContents instanceof Widget) {
+                    tdContents.target(this);
+                    tdContents._parentWidget = context;
+                    if (tdContents._class.indexOf("childWidget") < 0) {
+                        tdContents._class += " childWidget";
+                    }
+                    tdContents.render();
+                    context.childWidgets().push(tdContents);
+                }
             });
             var dis = d3.select(this);
             context.applyStyleToRows(dis);
         });  
 
-        var tableMarginHeight = this.thead.node().offsetHeight;
+        var tableMarginHeight = this.thead.style("height");
 
-        var tbodyRows = context.table.select("tbody tr");
+        var tbodyRows = this.table.select("tbody").select("tr");
```

Comment:
I think he gets away with it because it will grab the first one - if it was SelectAll there would be trouble...